src/main/java/seedu/module/model/module/Deadline.java

```diff
@@ -2,25 +2,81 @@
 
 import static java.util.Objects.requireNonNull;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.EnumSet;
+
+import seedu.module.logic.commands.exceptions.CommandException;
+import seedu.module.logic.parser.exceptions.ParseException;
+
+/**
+ * Represents Priority tags in Enum
+ */
+enum Priority {
+    HIGH, MEDIUM, LOW;
+    public boolean isValid() {
+        return true;
+    }
+}
+
 /**
  * Represents a Module's deadline in the module book.
  * Guarantees: immutable; is always valid
  */
 public class Deadline {
     public static final String MESSAGE_CONSTRAINTS = "Not a valid Deadline";
+    protected Date date;
 
     private String description;
     private String time;
     private boolean isDone;
     private boolean isInProgress;
     private String tag;
-    private int deadlineListNum = 0;
 
-    public Deadline(String description, String time, String tag) {
+    public Deadline (String description, String time, String tag) throws ParseException, CommandException {
         requireNonNull(description);
+        int maxLength = 70;
+        if (description.length() > maxLength) {
+            description = description.substring(0, maxLength);
+        }
```

Comment:
[MINOR] Would this possibly be an inconvenience to the user? Maybe you can consider doing some text overflow on the Ui to counter super long descriptions?

Reply:
Just to clarify, this is because the window has variable width. So maxing the length of the description to 70 may just seem like a weird constraint if it doesn't affect users using the application with a large width.